File: internal/parser/swiftparser/testdata/sample.swift
import Foundation
import UIKit

// Top-level function
func topLevelFunc(x: Int) -> String {
    return String(x)
}

// Top-level async throwing function
func fetchData(from url: URL) async throws -> Data {
    return Data()
}

// Top-level properties
var globalVar: Int = 42
let globalLet: String = "hello"

// Type aliases
typealias Completion = (Result<Data, Error>) -> Void
typealias StringDict = [String: Any]

// Class with inheritance
class UserService: BaseService {
    var name: String
    let id: Int

    func getUser(id: Int) -> User {
        return User(name: "", email: "")
    }

    static func create() -> UserService {
        return UserService()
    }

    private func internalMethod() {}

    init(name: String) {
        self.name = name
        self.id = 0
    }
}

// Struct conforming to protocol
struct User: Codable, Hashable {
    let name: String
    let email: String

    func displayName() -> String {
        return name
    }
}

// Protocol with associated type
protocol DataSource {
    associatedtype Item
    func fetchAll() -> [Item]
    var count: Int { get }
}

// Protocol with inheritance
protocol Configurable: AnyObject {
    func configure(with options: [String: Any])
}

// Enum with associated values
enum APIError: Error {
    case notFound
    case unauthorized
    case custom(message: String)
}

// Extension
extension String {
    func trimmed() -> String {
        return trimmingCharacters(in: .whitespaces)
    }

    var isBlank: Bool {
        return isEmpty
    }
}

// Actor
actor DataStore {
    var items: [String] = []

    func add(item: String) {
        items.append(item)
    }
}

// Generic class
class Repository<T: Codable> {
    func findById(id: String) -> T? {
        return nil
    }

    func save(entity: T) {}
}

// Nested types
class Outer {
    struct Inner {
        let value: Int
    }

    enum Direction {
        case up, down, left, right
    }
}

// Private top-level (should not appear)
private func privateHelper() {}
fileprivate class InternalOnly {}
private var _secret: String = "hidden"

// Open class
open class BaseComponent {
    open func render() -> String { return "" }
}

File: internal/parser/kotlinparser/testdata/sample.kt
package com.example

import kotlinx.coroutines.flow.Flow

// Top-level function
fun topLevelFun(x: Int): String {
    return x.toString()
}

// Suspend function
suspend fun fetchData(url: String, timeout: Int = 30): String {
    return ""
}

// Top-level properties
val appName: String = "MyApp"
var retryCount: Int = 3
const val MAX_RETRIES: Int = 5

// Type aliases
typealias StringMap = Map<String, String>
typealias Callback<T> = (T) -> Unit

// Data class
data class User(val name: String, val email: String)

// Sealed class with generics
sealed class Result<T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Error(val message: String) : Result<Nothing>()
}

// Abstract class
abstract class BaseService {
    abstract fun initialize(): Unit
    open fun configure() {}
    private fun internalSetup() {}
}

// Regular class with inheritance
class UserService : BaseService() {
    fun getUser(id: Int): User {
        return User("", "")
    }

    override fun initialize() {}

    companion object {
        fun create(): UserService = UserService()
        const val TAG: String = "UserService"
    }
}

// Interface with generics
interface Repository<T> {
    fun findById(id: String): T?
    fun save(entity: T): Unit
    val count: Int
}

// Object declaration (singleton)
object NetworkModule {
    fun provideClient(): Any = Any()
    val baseUrl: String = "https://api.example.com"
}

// Enum class
enum class Status {
    ACTIVE, INACTIVE, SUSPENDED;
    fun isActive(): Boolean = this == ACTIVE
}

// Annotation class
annotation class Inject

// Generic class with constraint
class GenericRepo<T : Any>(private val items: MutableList<T> = mutableListOf()) {
    fun findAll(): List<T> = items
    fun add(item: T) { items.add(item) }
}

// Inline function
inline fun <reified T> fromJson(json: String): T {
    throw NotImplementedError()
}

// Extension function (top-level)
fun String.capitalize(): String {
    return replaceFirstChar { it.uppercase() }
}

// Class with named companion object
class Config {
    companion object Factory {
        fun default(): Config = Config()
    }
}

// Private top-level (should not appear)
private fun privateHelper(): Unit {}
private class _InternalClass {}
private val _secret: String = "hidden"

// Internal top-level (should not appear)
internal fun internalHelper(): Unit {}
internal class InternalOnlyClass {}
